x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/checkpoint/DefaultCheckpointProvider.java

```diff
@@ -234,10 +239,7 @@ private static void getCheckpointsFromOneClusterV2(
         );
         ActionListener<GetCheckpointAction.Response> checkpointListener;
         if (RemoteClusterService.LOCAL_CLUSTER_GROUP_KEY.equals(cluster)) {
-            checkpointListener = ActionListener.wrap(
-                checkpointResponse -> listener.onResponse(checkpointResponse.getCheckpoints()),
-                listener::onFailure
-            );
+            checkpointListener = listener.delegateFailure((l, r) -> l.onResponse(r.getCheckpoints()));
```

Comment:
Would it make sense to keep using the more descriptive name (`checkpointResponse`) rather than `r`?

Reply:
Yes, though I replaced this with `listener.safeMap(GetCheckpointAction.Response::getCheckpoints)` since it will do the `l.onResponse` for us